queries/dart/highlights.scm

```diff
@@ -0,0 +1,104 @@
+(dotted_identifier_list) @string
+
+; Methods
+
+(function_type
+  name: (identifier) @method)
+(super) @function
+
+; Annotations
+; TODO: highlight??
+(annotation
+  name: (identifier) @attribute)
+(marker_annotation
+  name: (identifier) @attribute)
+
+; TODO: operators
+[
+ "@" 
+ (additive_operator)
+] @operator
+; TODO: delimiers/punctuation
+
+; Types
+
+(class_definition
+  name: (identifier) @type)
+(constructor_signature
+  name: (identifier) @type)
+(type_identifier
+  (identifier) @type)
+(scoped_identifier
+  scope: (identifier) @type)
+(function_signature
+  name: (identifier) @method)
+(enum_declaration
+  name: (identifier) @type)
+(enum_constant
+  name: (identifier) @type)
+(type_identifier) @type
+(void_type) @type
+
+((scoped_identifier
+  scope: (identifier) @type
+  name: (identifier) @type)
+ (#match? @type "^[a-zA-Z]"))
+
+(type_identifier) @type
+
+; Variables
+
+((identifier) @constant
+ (#match? @constant "^_*[A-Z][A-Z\d_]+"))
+
+(this) @constant.builtin
+
+;; Parameters
+
+(formal_parameter
+    name: (identifier) @parameter)
+
+;; Literals
+
+(hex_integer_literal) @number
+(decimal_integer_literal) @number
+(decimal_floating_point_literal) @float
+;; Those node do not seem to exist
+;(octal_integer_literal) @number
+;(hex_floating_point_literal) @float
+
+(string_literal) @string
+(true) @boolean
+(false) @boolean
+(null_literal) @constant.builtin
+
+(comment) @comment
+
+; Keywords
+
+["import" "library" "export"] @include
+
+"abstract" @keyword
+"assert" @keyword
+"break" @keyword
+"on" @keyword
+"class" @keyword
+"default" @keyword
+"enum" @keyword
+"extends" @keyword
+"final" @keyword
+"implements" @keyword
+"is" @keyword
+"as" @keyword
+"mixin" @keyword
+"external" @keyword
+"new" @keyword
+"return" @keyword
+"static" @keyword
+; "rethrow" @keyword
```

Comment:
This is a valid keyword much like the ones above not sure if it's causing some kind of error but just FYI

Reply:
as well as "async", "var" "await" they need to be captured as something different. Would be happy if someone could take over who is actually using dart.

Reply:
@theHamsta is it worth merging what you have so far (obviously up to you and the other maintainers) and I can make a new branch later to add some more stuff so the surface area of my PR is smaller. I think what this PR has is already going to be really useful